trollflow2/plugins/__init__.py

```diff
@@ -302,16 +324,106 @@ def save_datasets(job):
     base_config = job['input_mda'].copy()
     base_config.pop('dataset', None)
     eager_writing = job['product_list']['product_list'].get("eager_writing", False)
-    with renamed_files() as renames:
+    early_moving = job['product_list']['product_list'].get("early_moving", False)
+    sentinel = object()
+    call_on_done = job["product_list"]["product_list"].get("call_on_done", sentinel)
+    if call_on_done is not sentinel:
+        callbacks = [dask.delayed(c) for c in call_on_done]
```

Comment:
It could.  I don't feel very strongly either way.  My reason: accepting None makes the API slightly more complicated, as we then accept two very different argument types (sequence or None), which would need to be marked wherever we document valid values for `call_on_done`, as well as in tests.  By defining a sentinel inside the function, passing None is invalid and needs to be neither documented nor tested.

Reply:
Changed to using `None` in 6bd0218